index.html

```diff
@@ -698,17 +699,17 @@
   for (let record of message.records) {
     switch (record.recordType) {
       case "text":
-        console.log(`Text: ${record.data()}`);
+        console.log(`Text: ${record.toText()}`);
         break;
       case "url":
-        console.log(`URL: ${record.data()}`);
+        console.log(`URL: ${record.toText()}`);
         break;
       case "json":
-        console.log(`JSON: ${record.data().myProperty.toString()}`);
+        console.log(`JSON: ${record.toJSON().myProperty.toString()}`);
         break;
       case "opaque":
         if (record.mediaType.startsWith('image/') {
-          const blob = new Blob([record.data()], {type: record.mediaType});
+          const blob = new Blob([record.toArrayBuffer()], {type: record.mediaType});
```

Comment:
do you still need the [] here? you are turning an array buffer into an array and then into a blob

Reply:
Yes, we still need it, as the first argument of Blob constructor should be `optional sequence<BlobPart> blobParts`, see https://w3c.github.io/FileAPI/#dfn-Blob